### ISSUE TITLE
Merge Syndic branch into Master (aka switch to new Planet)

### ISSUE THREAD

**Issue by amirmc** [ocaml/v2.ocaml.org#569]
Hi folks,

It seems things are largely settled on the syndic branch so I'd like to surface any blockers to getting it merged into the live site.  I'm aware of one outstanding issue to do with Jane Street's feed but that may have been fixed already.

Once it's up and running I'd then like to get help to [remove the unix dependency](https://github.com/Cumulus/Syndic/issues/23#issuecomment-56914187) so that both [Mirage](https://github.com/mirage/mirage-www/issues/156) and [Nymote](http://nymote.org) sites can also begin aggregating feeds.

Can we put together a checklist?

**Comment by Chris00** [comment]
- I have to implement the resolution of image URLs (as in https://github.com/ocaml/platform-blog/issues/12#issuecomment-52939811).  The aggregated feed contains enough information to do that but I haven't found the time to do it yet... :-(
- There is also the [pagination of news](https://github.com/ocaml/ocaml.org/issues/374).  I have not decided yet whether the following page should just be a list of headlines or whether to create several pages (including _all_ older news?).

It would be nice if a few persons tried that branch and reported what does not seem right to them.

**Comment by Chris00** [comment]
For the Unix dependency, I filed a [feature wish for Calendar](https://forge.ocamlcore.org/tracker/index.php?func=detail&aid=1449&group_id=83&atid=418).

**Comment by Chris00** [comment]
I have [removed the use of Unix](https://github.com/Cumulus/Syndic/commit/997afc75afd2b21ebd7dd1cdd66de1723fe3bf84) in Syndic.

**Comment by Chris00** [comment]
So, if calendar is split, we can get rid of the Unix dependency.

**Comment by Chris00** [comment]
Merged `xml:base` resolution.

**Comment by Chris00** [comment]
I rendered the older posts as a simple list.
@amirmc Is it good enough or do we want something more sophisticated.

**Comment by Chris00** [comment]
More `xml:base` resolution (of stories).  I also generate a `feed.xml` and `opml.xml`.  I think we are ready to merge — it would be better that a new version of Syndic is released tough.
@dinosaure ?

**Comment by Chris00** [comment]
@amirmc ?

**Comment by amirmc** [comment]
Sorry for the delay and thanks for all the work!  I think a simple list for older posts is fine and we easily can revisit that decision if needed.

I think it's better to use released versions of packages so cutting a new version of Syndic would be preferable.  How soon could that happen @dinosaure? If there's likely to be a delay then I'd favour just merging instead.

I'll try and see what we can get done about the Unix dependency in Calendar.

**Comment by Chris00** [comment]
On Fri, 9 Jan 2015 13:35:38 -0800, Amir Chaudhry wrote:

> Sorry for the delay and thanks for all the work! I think a simple list for
> older posts is fine and we easily can revisit that decision if needed.

I splitted the list per month.

> I think it's better to use released versions of packages so cutting a new
> version of Syndic would be preferable. How soon could that happen @dinosaure?
> If there's likely to be a delay then I'd favour just merging instead.

I could do it but I prefer a green light of @dinosaure

> I'll try and see what we can get done about the Unix dependency in Calendar.

OK

**Comment by Chris00** [comment]
Well, I know that @dinosaure has problems with his computer so I did a [new release](https://github.com/Cumulus/Syndic/releases/tag/v1.2) — [submitted](https://github.com/ocaml/opam-repository/pull/3343) to OPAM.

**Comment by Chris00** [comment]
I believe the only "blocker" to merge is the release of [up-to-date `opamfu` and `opam2web` packages](https://github.com/ocaml/ocaml.org/issues/609).

**Comment by dsheets** [comment]
Clearing that is now my primary task.

**Comment by Chris00** [comment]
@dsheets thanks

**Comment by dsheets** [comment]
I think you're unblocked on those issues now.

**Comment by amirmc** [comment]
w00t! Nearly there!  At some point, I'll write a brief blog post to highlight these efforts and also try to draw some attention to the remaining (but separate) issue (https://github.com/Cumulus/Syndic/issues/23#issuecomment-56914187). @Chris00 Do you think it would be reasonable to summarise the work needed for that so that we can add it to the [Pioneer Projects](https://github.com/mirage/mirage-www/wiki/Pioneer-Projects)?

**Comment by Chris00** [comment]
The new feed generation is now _live_! 🍷 We may want to truncate the aggragted Atom feed however, it is 7.3M at the moment... (I do not use a RSS reader, please tell me how many stories you would like to have access too.)

**Comment by agarwal** [comment]
Congratulations!!

**Comment by amirmc** [comment]
:+1:
:100:

**Comment by Drup** [comment]
> The new feed generation is now live! :wine_glass: We may want to truncate the aggragted Atom feed however, it is 7.3M at the moment... (I do not use a RSS reader, please tell me how many stories you would like to have access too.)

One month ? Is it small enough ? Any decent RSS feed reader will archive things.

**Comment by Chris00** [comment]
I have [limited the feed to the last 50 posts](https://github.com/ocaml/ocaml.org/commit/ebf6a888759349088989999813ffc744ab6e1067) so it now has a size of about 600–700Kb.
